lib/browser/tag/tag.js

```diff
@@ -116,24 +116,28 @@ function Tag(impl, conf, innerHTML) {
   defineProperty(this, 'mixin', function() {
     each(arguments, function(mix) {
       var instance
+      var props = []
+      var obj
 
       mix = typeof mix === T_STRING ? riot.mixin(mix) : mix
 
       // check if the mixin is a function
       if (isFunction(mix)) {
         // create the new mixin instance
         instance = new mix()
-        // save the prototype to loop it afterwards
-        mix = mix.prototype
       } else instance = mix
 
+      // build multilevel prototype inheritance chain property list
+      do props = props.concat(Object.getOwnPropertyNames(obj || instance));
+      while (obj = Object.getPrototypeOf(obj || instance));
+
       // loop the keys in the function prototype or the all object keys
-      each(Object.getOwnPropertyNames(mix), function(key) {
```

Comment:
I would prefer looping only on the enumerable properties of the mixin objects. Why have you removed it?